src/network/ClientConnectionManager.ts

```diff
@@ -632,39 +639,38 @@ export class ClientConnectionManager extends EventEmitter {
-                                + 'the client does not match the one configured in the cluster or '
-                                + 'the credentials set in the client security config could not be authenticated');
-                            break;
-                        case AuthenticationStatus.SERIALIZATION_VERSION_MISMATCH:
-                            error = new IllegalStateError('Server serialization version does not match to client.');
-                            break;
-                        case AuthenticationStatus.NOT_ALLOWED_IN_CLUSTER:
-                            error = new ClientNotAllowedInClusterError('Client is not allowed in the cluster');
-                            break;
-                        default:
-                            error = new AuthenticationError('Authentication status code not supported. Status: '
-                                + response.status);
-                    }
-                    connection.close('Failed to authenticate connection', error);
-                    return Promise.reject(error);
+        return timedPromise(
+            this.client.getInvocationService().invokeUrgent(invocation),
+            this.authenticationTimeout
+        ).catch((e) => {
+            connection.close('Failed to authenticate connection', e);
+            throw e;
+        }).then((responseMessage) => {
+            const response = ClientAuthenticationCodec.decodeResponse(responseMessage);
+            if (response.status === AuthenticationStatus.AUTHENTICATED) {
+                this.handleSuccessfulAuth(connection, response);
+            } else {
+                let error: Error;
+                switch (response.status) {
+                    case AuthenticationStatus.CREDENTIALS_FAILED:
+                        error = new AuthenticationError('Authentication failed. The configured cluster name on '
+                            + 'the client does not match the one configured in the cluster or '
+                            + 'the credentials set in the client security config could not be authenticated');
+                        break;
+                    case AuthenticationStatus.SERIALIZATION_VERSION_MISMATCH:
+                        error = new IllegalStateError('Server serialization version does not match to client.');
+                        break;
+                    case AuthenticationStatus.NOT_ALLOWED_IN_CLUSTER:
+                        error = new ClientNotAllowedInClusterError('Client is not allowed in the cluster');
+                        break;
+                    default:
+                        error = new AuthenticationError('Authentication status code not supported. Status: '
+                            + response.status);
                 }
-            });
+                connection.close('Failed to authenticate connection', error);
+                return Promise.reject(error);
```

Comment:
Minor: Can be replaced with `throw error`

Reply:
Good point. That return was in the original code and it didn't catch my eye. Addressed in 9994e713ce2c4fae6b1008b218e856f7b428a97e